crates/engine/tree/src/download.rs

```diff
@@ -29,6 +46,241 @@ pub enum DownloadOutcome {
     Blocks(Vec<SealedBlockWithSenders>),
 }
 
+/// A wrapper type around [`SealedBlockWithSenders`] that implements the [Ord]
+/// trait by block number.
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct OrderedSealedBlockWithSenders(SealedBlockWithSenders);
```

Comment:
can we move this helper type below the BasicDownloader impl?

Reply:
sure, done